### PR TITLE
Alt Router: Support aux routes/RouterLink/Location

### PR DESCRIPTION
**Pull request by vsavkin**

### CODE REVIEW

File: modules/angular2/src/alt_router/router_url_parser.ts [hunk 1]
```diff
@@ -8,20 +10,157 @@ export class DefaultRouterUrlParser extends RouterUrlParser {
...
+
+    params[key] = value;
+  }
+
+  parseAuxiliaryRoutes(): TreeNode<UrlSegment>[] {
+    var segments = [];
+    this.capture('(');
+
+    while (!this.peekStartsWith(')') && this._remaining.length > 0) {
+      segments = segments.concat(this.parseSegments("aux"));
```
Comment (brandonroberts) [inline modules/angular2/src/alt_router/router_url_parser.ts]: Make "aux" an overridable default?

File: modules/angular2/src/alt_router/directives/router_link.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+export class RouterLink implements OnDestroy {
+  @Input() target: string;
+  private _changes: any[] = [];
+  private _targetUrl: Tree<UrlSegment>;
+  private _subscription: any;
+
+  @HostBinding() private href: string;
+
+  constructor(private _router: Router, private _segment: RouteSegment) {
+    this._subscription = ObservableWrapper.subscribe(_router.changes, (_) => {
```
Comment (vsavkin) [inline modules/angular2/src/alt_router/directives/router_link.ts]: Remove the wrapper

File: modules/angular2/test/alt_router/router_url_serializer_spec.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+      let tree = url.parse("/one;a");
+
+      let c = tree.firstChild(tree.root);
+      expectSegment(c, "one");
+      expectSegment(tree.firstChild(c), ";a=true");
+
+      expect(url.serialize(tree)).toEqual("/one;a=true");
+    });
+
+    // it("should parse key-value query params", () => {
```
Comment (vsavkin) [inline modules/angular2/test/alt_router/router_url_serializer_spec.ts]: Remove the comments

**Comment by vsavkin** [comment]
Did a walkthrough with Misko

**Comment by mprobst** [comment]
Woohoo @vsavkin !

**Pull request closed without merging**